plugins/woocommerce/src/Admin/API/ProductsLowInStock.php

```diff
@@ -216,16 +258,15 @@ protected function transform_post_to_api_response( $query_result ) {
 	 * Generate a query.
 	 *
 	 * @param bool $siteside_only generates a query for sitewide low stock threshold only query.
+	 * @param bool $return_count_query returns country query instead of the actual query.
 	 *
 	 * @return string
 	 */
-	protected function get_query( $siteside_only = false ) {
+	protected function get_query( $siteside_only = false, $return_count_query = false ) {
```

Comment:
Having this as a parameter feels very strange to me. I would prefer a `get_count_query` function. Although that may require some extra refactor to avoid a large amount of duplicate code, which I assume is some of the reason you stuck to this approach.
Maybe we could separate it out into `get_post_meta_args` and `get_base_query` ( or something to that extend ).

This would of been a lot easier if we made use of [WP_Query](https://developer.wordpress.org/reference/classes/wp_query/).

Reply:
Yup..that was the reason.

I actually tried to use WP_Query when I first worked on this endpoint last year and I could not find a good way to work with WP_Query. I'll look into it one more time to see if we can use WP_Query.

Reply:
@louwie17 I've refactored code in https://github.com/woocommerce/woocommerce/pull/40289/commits/991ddaa56d6d78ec383338c2542e9767e1cd5810
I wish there was a generic query builder in WP or WC (May be a cooldown/HACK issue :) ? )

Reply:
Hey @louwie17 ✋ Could you take another look at it please?

Reply:
Yup, sorry for getting back to this so late, I sometimes loose track of my GH notifications.